File: leetcode_1400/ConstructKPalindromeStrings.java
package leetcode_1400;

class ConstructKPalindromeStrings{
    public boolean canConstruct(String s, int k) {
        if (k > s.length()) return false;

        int[] freq = new int[26];
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        int odd = 0;
        for (int f : freq) {
            if (f % 2 != 0) {
                odd++;
            }
        }
        return odd <= k;
    }

    public static void main(String[] args) {
        ConstructKPalindromeStrings ckps = new ConstructKPalindromeStrings();
        System.out.println(ckps.canConstruct("annabelle", 2));
        System.out.println(ckps.canConstruct("leetcode", 3));
    }
}

File: leetcode_1400/ConstructKPalindromeStrings.py
from typing import Counter


class ConstructKPalindromeStrings:
    def canConstruct(self, s: str, k: int) -> bool:
        if k > len(s):
            return False

        # Count the frequency of each character
        freq = Counter(s)
        odd = 0
        for count in freq.values():
            if count % 2:
                odd += 1

        return odd <= k

def main():
    v = ConstructKPalindromeStrings()
    s = "annabelle"
    k = 2
    print(v.canConstruct(s, k))

if __name__ == "__main__":
    main()